### DISCUSSION TITLE
Delete QR code scanner images

### DISCUSSION THREAD

**Discussion by m-wagner98** [capawesome-team/capacitor-mlkit#189] [Q&A] [answered]
Hey community just wanted to ask whether anyone knows whether the capacitor-mlkit Barcode scanner saves images of the scans on the device (local app data). Some QR codes may contain sensitive data. In this case they would be stored unencrypted on the device which was a finding in our last Penetration test report. Does anyone know whether 

1. the app persists scan images
2. there is a way to specify to not do it (delete them directly after scan)

Any help appreciated.

**Comment by robingenz** [comment, marked as answer]
Hey, this is easy: The plugin does not persist any data. The plugin just reads the QR code and returns the value. Besides that, make sure to take a look at the terms & privacy sections of the integrated ML Kit SDK: https://capawesome.io/plugins/mlkit/barcode-scanning/#terms-privacy

**Comment by m-wagner98** [comment]
Thanks a lot!
